web/src/utils/helpers.ts

```diff
@@ -15,21 +20,26 @@ export function getCO2IntensityByMode(
 export function dateToDatetimeString(date: Date) {
   return date.toISOString().split('.')[0] + 'Z';
 }
-
-export function getProductionCo2Intensity(mode: string, zoneData: any) {
+export function getProductionCo2Intensity(
+  mode: ElectricityModeType,
+  zoneData: ZoneDetail
+) {
   const isStorage = mode.includes('storage');
-  const resource = mode.replace(' storage', '');
+  const generationMode = mode.replace(' storage', '') as GenerationType;
 
-  const storage = (zoneData.storage || {})[resource];
-  const storageCo2Intensity = (zoneData.storageCo2Intensities || {})[resource];
-  const dischargeCo2Intensity = (zoneData.dischargeCo2Intensities || {})[resource];
-  const productionCo2Intensity = (zoneData.productionCo2Intensities || {})[resource];
+  if (!isStorage) {
+    return zoneData.productionCo2Intensities?.[generationMode];
+  }
 
-  if (isStorage) {
-    return storage > 0 ? storageCo2Intensity : dischargeCo2Intensity;
+  const storageMode = generationMode as ElectricityStorageKeyType;
+  const storage = zoneData.storage?.[storageMode];
+  // If storing, we return 0 as we don't want to count it as CO2 emissions until electricity is discharged.
+  if (storage && storage > 0) {
+    return 0;
   }
 
```

Comment:
Made this change here, was waiting for feedback in https://github.com/electricitymaps/electricitymaps-contrib/issues/3668#issuecomment-1445826251 :)